pkg/odo/genericclioptions/runnable.go

```diff
@@ -38,25 +42,35 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 	// Prompt the user to consent for telemetry if a value is not set already
 	// Skip prompting if the preference command is called
 	// This prompt has been placed here so that it does not prompt the user when they call --help
+
 	if !cfg.IsSet(preference.ConsentTelemetrySetting) && cmd.Parent().Name() != "preference" {
-		var consentTelemetry bool
-		prompt := &survey.Confirm{Message: "Help odo improve by allowing it to collect usage data. Read about our privacy statement: https://developers.redhat.com/article/tool-data-collection. You can change your preference later by changing the ConsentTelemetry preference.", Default: false}
-		err = survey.AskOne(prompt, &consentTelemetry, nil)
-		ui.HandleError(err)
-		if err == nil {
-			if err1 := cfg.SetConfiguration(preference.ConsentTelemetrySetting, strconv.FormatBool(consentTelemetry)); err1 != nil {
-				klog.V(4).Info(err1.Error())
+		if !segment.RunningInTerminal() {
+			klog.V(4).Infof("Skipping telemetry question because there is no terminal (tty)\n")
+		} else if os.Getenv(disableTelemetryEnv) == "true" {
```

Comment:
There are in all four calls to `os.Getenv` in this function. Would it make sense to store the value in a variable and reduce it to just one call? TBH, I don't know how costly a call like `os.Getenv` would be from performance point of view.

Reply:
the call is cheap, but it might be more readable if it is stored in one call.